api/python/quilt3/packages.py

```diff
@@ -1262,7 +1262,7 @@ def _set(
             entry = PackageEntry(write_pk, size, hash_obj=None, meta=new_meta)
 
         else:
-            raise TypeError(f"Expected a string for entry, but got an instance of {type(entry)}.")
+            raise TypeError(f"Expected a string for entry, but got an instance of {type(entry)}: {entry}.")
```

Comment:
is there a chance string formatting will barf here bc the object isnt stringifiable or smth?

Reply:
Fair. Reverting